src/CMakeLists.txt

```diff
@@ -350,6 +350,11 @@ else ()
         SOVERSION "${NN_ABI_VERSION}")
 endif ()
 
+# Set library outputs same as top-level project binary outputs
+set_target_properties (${PROJECT_NAME} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR})
+set_target_properties (${PROJECT_NAME} PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR})
+set_target_properties (${PROJECT_NAME} PROPERTIES ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR})
+
```

Comment:
@Snaipe, @gdamore, @shiva what do you think about this fix?

It's to resolve a linking problem on Windows that does not occur on *NIX systems. This bug is what has been causing AppVeyor to timeout and fail in all the tests on this PR.

Presumably, this bug is manifested from a bad interaction between `add_subdirectory` for `src/CMakeScripts.txt` and this output behavior unique to MSVC: http://stackoverflow.com/questions/7747857/in-cmake-how-do-i-work-around-the-debug-and-release-directories-visual-studio-2

It solves the problem by dumping the output `nanomsg.dll` (or `nanomsg.lib` for static libs) into the binary location specified in the parent `CMakeLists.txt` -- which is incidentally where all the test executables end up.

Even though this fix affects *NIX systems, I think it cleans up the structure of the artifacts a bit nicer (opinion), and for this reason didn't wrap this block in a `if (MSVC)` statement.

The other reason for doing it like this is to scope changes as narrowly as possible -- unlike the previous `src/CMakeScripts.txt` still currently in mainline that changed globally-scoped `CMAKE_...` variables -- https://github.com/nanomsg/nanomsg/blob/master/src/CMakeLists.txt#L24-L26

Thoughts on this patch? Any gotchas? Or a better way?

---

And two semi-related ideas I had while solving this issue:
1. collapse `src/CMakeScripts.txt` into `CMakeLists.txt`, obviating need for `add_subdirectory` and its scoping headaches (especially, in the bigger context of when nanomsg is a subproject of another project), or ...
2. ... going the opposite direction, and creating two more `CMakeLists.txt` for both the `tests` and `tools` directory, both for overall consistency, and also to meet composability problems such as these head-on by setting precedent of best practices on how to do so.

Thoughts here?